src/pages/profile/VerifyFacebook.js

```diff
@@ -70,22 +77,47 @@ class VerifyFacebook extends Component {
     )
   }
 
+  clearErrors() {
+    this.setState({generalErrors:[]})
+  }
+
   onCertify() {
-    var w = window.open(this.state.url, '', 'width=650,height=500')
+    this.clearErrors()
+    var fbWindow = window.open(this.state.url, '', 'width=650,height=500')
 
-    const finish = e => {
-      var data = String(e.data)
+    const finish = async e => {
+      const data = String(e.data)
       if (!data.match(/^origin-code:/)) {
         return
       }
       window.removeEventListener('message', finish, false)
-      if (!w.closed) {
-        w.close()
+      if (!fbWindow.closed) {
+        fbWindow.close()
       }
 
-      origin.attestations
-        .facebookVerify({ code: data.split(':')[1] })
-        .then(result => this.props.onSuccess(result))
+      try {
+        const facebookAttestation = await origin.attestations
+          .facebookVerify({ code: data.split(':')[1] })
+
+        this.props.onSuccess(facebookAttestation)
+      } catch (exception) {
+        const errorsJson = JSON.parse(exception).errors
+          
+        // Service exceptions --> general error
+        if (Array.isArray(errorsJson))
+          this.setState({generalErrors: errorsJson})
+        // Form field error. Since no fields are displayed in the DAPP convert form field errors to general errors
+        else
+          this.setState({
+            generalErrors: Object.keys(errorsJson)
+              // Prepend the error with the field that is causing the error
+              .map(field =>
+                field + ": " +
+                // Join all the errors into a single string
+                errorsJson[field].join(' ')
+              )
+          })
```

Comment:
This is cool 👍 
Template literal may clean it up a bit (again, missing the backticks):
`.map(field => ${field} : ${errorsJson[field].join(' ')})`